solutions/gold/usaco-862.mdx

```diff
@@ -0,0 +1,105 @@
+---
+id: usaco-862
+source: USACO Gold 2018 December
+title: Cowpatibility
+author: Ryan Chou
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_cowpatibility_gold_dec18.html)
+
+<Spoiler title="Hint">
+
+Instead of trying to find the number of incompatible pairs, try to find the number of *compatible* pairs instead!
+
+This is known as [complementary counting](https://artofproblemsolving.com/wiki/index.php/Complementary_counting).
+
+</Spoiler>
+
+<Spoiler title="Solution">
+
+## Explanation
+
+Since there are $\frac{N(N-1)}{2}$ total pairs of cows, if we can count the numbers of compatible pairs, we can just subtract it from the total to get our answer!
+
+We can use [PIE (Principle of Inclusion/Exclusion)](https://artofproblemsolving.com/wiki/index.php/Principle_of_Inclusion-Exclusion) to calculate this! This means we'll have to add all pairs of cows that share one flavor in common, subtract cows that share two flavors,
+and so on...
+
+To understand why this works, imagine that we only have three flavors of cows, and that some cows share more than one flavor with each other. Then, if we were to only
+count the pairs of cows with one flavor, the pairs of cows that share two flavors would be overcounted! See the animation below for a visualization.
+
+<img src = "/usaco-862-anim/PIE.gif"/>
+
+In order to avoid considering rearranged sets of the same flavors, we'll sort the favorite flavors of each cow beforehand.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
```

Comment:
I think there's a log factor in your solution from the map

Reply:
ur right, thanks <3